asv_bench/benchmarks/groupby.py

```diff
@@ -510,6 +511,27 @@ def time_groupby_extra_cat_nosort(self):
         self.df_extra_cat.groupby("a", sort=False)["b"].count()
 
 
+class CategoricalFrame:
+    def setup(self):
+        SIZE = 100000
+        GROUPS = 10000  # The larger, the more extreme the timing differences
+        CARDINALITY = 10
+        CAT = CategoricalDtype(list(range(CARDINALITY)))
+        df_int = DataFrame(
+            {
+                "group": [np.random.randint(0, GROUPS) for i in range(SIZE)],
+                "cat": [np.random.choice(CAT.categories) for i in range(SIZE)],
+            }
+        )
+        self.df_cat_values = df_int.astype({"cat": CAT})
+
+    def time_groupby(self):
```

Comment:
likely need to paramaterize this over a bunch of functions. assume this is not too slow to do that. if so pls reduce the size of the benchmark to make it reasonable to parameterize.

Reply:
this takes 10.2 ms ± 28.7 µs per loop on the current commit on my desktop via timeit but 7.14 s ± 31.3 ms per loop on master.  I think 10ms should be ok but please let me know otherwise

Reply:
yes this is fine, but i need you to paramaterize over multiple functions, doesn't have to be every one but represententative ones (e.g. reductions, transforms and filters)

Reply:
paramaterized over column types in b4648d5a0